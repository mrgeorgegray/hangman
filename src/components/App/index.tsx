import React, { Fragment } from "react";

import { useGameState } from "../../context/Game";
import Game from "../../views/Game";
import Home from "../../views/Home";
import Header from "../Header";
import Footer from "../Footer";

const App: React.FC = () => {
  const { status } = useGameState();

  return (
    <Fragment>
      <Header />
      <main>{status === "notStarted" ? <Home /> : <Game />}</main>
      <Footer />
    </Fragment>
  );
};

export default App;
