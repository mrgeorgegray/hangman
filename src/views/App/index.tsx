/** @jsxImportSource @emotion/react */

import React from "react";
import { css, Global, useTheme } from "@emotion/react";
import emotionNormalize from "emotion-normalize";

import { useGameDispatch, useGameState } from "../../context/Game";
import Game from "../Game";
import Home from "../Home";
import SelectTopic from "../SelectTopic";
import Footer from "../../components/Footer";
import ThemeSwitch from "../../components/ThemeSwitch";

const App = () => {
  const { breakpoints, colors, fontSize, layout, space } = useTheme();
  const { status, theme, topic } = useGameState();
  const dispatch = useGameDispatch();

  function handleThemeChange() {
    if (theme === "dark") {
      dispatch({ type: "setTheme", payload: { theme: "light" } });
    } else {
      dispatch({ type: "setTheme", payload: { theme: "dark" } });
    }
  }

  return (
    <React.Fragment>
      <Global
        styles={css`
          ${emotionNormalize}
          * {
            box-sizing: border-box;
            -webkit-font-smoothing: antialiased;
          }
          html,
          body {
            background-color: ${colors.background};
            color: ${colors.text};
            font-family: monospace;
            font-size: ${fontSize[1]}px;
            line-height: 1.4;
          }
          a {
            color: ${colors.link};

            &:hover {
              color: ${colors.linkHover};
            }
          }
        `}
      />
      <div
        css={css`
          display: flex;
          flex-direction: column;
          margin: 0 auto;
          min-height: 100vh;
          max-width: ${layout.maxWidth};

          @media (min-width: ${breakpoints.lg}) {
            padding-top: ${space[5]}px;
          }
        `}
      >
        <main
          role="main"
          css={css`
            flex-grow: 1;
          `}
        >
          {status === "notStarted" && <Home />}
          {status !== "notStarted" && !topic && <SelectTopic />}
          {status !== "notStarted" && topic && <Game />}
        </main>
        <Footer />
        <ThemeSwitch currentTheme={theme} handleChange={handleThemeChange} />
      </div>
    </React.Fragment>
  );
};

export default App;
