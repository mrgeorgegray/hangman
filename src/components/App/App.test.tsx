import React from "react";
import { render } from "@testing-library/react";

import App from ".";
import GameContext from "../../context/Game";

describe("<App />", () => {
  it("renders home page when 'notStarted' ", () => {
    const { container } = render(
      <GameContext>
        <App />
      </GameContext>
    );

    expect(container).toMatchSnapshot();
  });

  it.todo("renders game");
});
