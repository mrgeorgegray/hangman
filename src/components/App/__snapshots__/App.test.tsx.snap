// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders home page when 'notStarted'  1`] = `
<div>
  <header>
    <p>
      Hangman App
    </p>
    <hr />
  </header>
  <main>
    <div>
      <h1>
        Welcome to Hangman!
      </h1>
      <p>
        Press the button to get started...
      </p>
      <button>
        Start Game
      </button>
    </div>
  </main>
  <footer
    style="position: fixed; left: 0px; bottom: 0px; width: calc(100% - 20px); padding: 10px;"
  >
    <hr />
    <p>
      Built with create-react-app.
    </p>
  </footer>
</div>
`;
